subsys/net/lib/dns/resolve.c

```diff
@@ -77,15 +82,73 @@ NET_BUF_POOL_DEFINE(dns_qname_pool, DNS_RESOLVER_BUF_CTR, DNS_MAX_NAME_LEN,
 
 static struct dns_resolve_context dns_default_ctx;
 
+static bool server_is_mdns(sa_family_t family, struct sockaddr *addr)
+{
+	if (family == AF_INET) {
+		if (net_is_ipv4_addr_mcast(&net_sin(addr)->sin_addr)) {
+			if (net_sin(addr)->sin_addr.s4_addr[3] == 251) {
+				return true;
+			} else {
```

Comment:
no need of that else {} here. Actually you could merge the 2 if in one with a && for returning true.